pkg/restore/restore.go

```diff
@@ -1217,13 +1218,13 @@ func (ctx *restoreContext) restoreItem(obj *unstructured.Unstructured, groupReso
 		return warnings, errs, itemExists
 	}
 
-	resourceClient, err := ctx.getResourceClient(groupResource, obj, namespace)
-	if err != nil {
-		errs.AddVeleroError(fmt.Errorf("error getting resource client for namespace %q, resource %q: %v", namespace, &groupResource, err))
-		return warnings, errs, itemExists
-	}
-
 	if groupResource == kuberesource.PersistentVolumes {
+		resourceClient, err := ctx.getResourceClient(groupResource, obj, namespace)
```

Comment:
Suggest replacing the `resourceClient` with the controller-runtime client in `handlePVHasNativeSnapshot()` function, this will reduce the confusion.
And the controller-runtime client is already a field(`kbClient`) of the `restoreContext`.

Reply:
I just move this code snippet from 1220-1225 to 1222-1227, such refactor should be done in another pr

Reply:
I can use `kbClient`, but it will introduce many code changes which are not directly  related to this pr